ipv8/test/test_configuration_pydantic.py

```diff
@@ -0,0 +1,250 @@
+import base64
+import enum
+import unittest
+
+from .base import TestBase
+from ..configuration import default
+
+try:
+    from ..configuration_pydantic import (BaseConfigurationClass, Bootstrapper, BootstrapperInit, DispersyBootstrapper,
+                                          DispersyBootstrapperInit, EdgeWalk, EphemeralKey, FileKey, IPv8Configuration,
+                                          Interface, Key, Overlay, PeriodicSimilarity, PingChurn, PreloadedKey,
+                                          RandomChurn, RandomWalk, UDPBroadcastBootstrapper,
+                                          UDPBroadcastBootstrapperInit, Walker, format_schema_recursive)
+
+    SKIP_TEST_CASE = False
+except ImportError:
+    SKIP_TEST_CASE = True
```

Comment:
This try-except block looks incorrect to me.

The intention of this block (as far as I understand) is skipping the tests in the case `pydantic` library is not installed.

In this case, the test suite prints the following message:

```
SKIPPED: Skipping optional pydantic tests because the library is not installed.
```

But a user will see the same error in case any of the import attempts fail. This behavior will hide the real problem.

The correct behavior here is to try to import something from the `pydantic` library and in the case of an error, sets  `SKIP_TEST_CASE = True`
```python

try:
    from pydantic import BaseModel

    SKIP_TEST_CASE = False
except ImportError:
    SKIP_TEST_CASE = True

from ..configuration_pydantic import (BaseConfigurationClass, Bootstrapper, BootstrapperInit, DispersyBootstrapper,
                                      DispersyBootstrapperInit, EdgeWalk, EphemeralKey, FileKey, IPv8Configuration,
                                      Interface, Key, Overlay, PeriodicSimilarity, PingChurn, PreloadedKey,
                                      RandomChurn, RandomWalk, UDPBroadcastBootstrapper, UDPBroadcastBootstrapperInit,
                                      Walker, format_schema_recursive)

```

Reply:
You are right.
This is the working example (tested with `pydantic`, without `pydantic` and with an import error):

```python
try:
    from pydantic import BaseModel

    SKIP_TEST_CASE = False
except ImportError:
    SKIP_TEST_CASE = True
else:
    from ..configuration_pydantic import (BaseConfigurationClass, Bootstrapper, BootstrapperInit, DispersyBootstrapper,
                                          DispersyBootstrapperInit, EdgeWalk, EphemeralKey, FileKey, IPv8Configuration,
                                          Interface, Key, Overlay, PeriodicSimilarity, PingChurn, PreloadedKey,
                                          RandomChurn, RandomWalk, UDPBroadcastBootstrapper,
                                          UDPBroadcastBootstrapperInit,
                                          Walker, format_schema_recursive)
```

Reply:
Done 👍

Reply:
That causes a Flake8 error: https://jenkins-ci.tribler.org/job/ipv8/job/test_ipv8_PR_flake8_check/1493//artifact/output/ipv8.test.test_configuration_pydantic.report.html

Reply:
Thanks!

Reply:
> F401: 'pydantic.BaseModel' imported but unused

Which is true. I'll check how to avoid or ignore this issue.

Reply:
I used importlib instead. It should do the same, without triggering Flake8.